test/unit/data-generator.js

```diff
@@ -0,0 +1,361 @@
+var expect = require('chai').expect,
+    objectAssign = require('object-assign'),
+    generate = require('../../lib/data-generator.js'),
+    EventEmitter = require('events').EventEmitter;
+
+var root = {
+      title: ''
+    },
+    suite = {
+      title: 'suite'
+    },
+    test = {
+      title: 'test',
+      ctx: {},
+      err: 'error'
+    },
+    rootSuite = {
+      title: '',
+      indent: 0,
+      tests: []
+    },
+    globalSuite = {
+      title: 'suite',
+      indent: 1,
+      tests: []
+    },
+    nestedSuite = {
+      title: 'suite',
+      indent: 2,
+      tests: []
+    },
+    passTest = {
+      title: 'test',
+      status: true,
+      result: undefined
+    },
+    failTest = {
+      title: 'test',
+      status: false,
+      error: 'error',
+      result: undefined
+    };
+
+/**
+ * Generates a natural strictly positive number
+ *
+ * @returns {Number}
+ */
+function getRandomNumber() {
+  return Math.floor(Math.random() * 100);
+}
+
+/**
+ * Generates an array of n random numbers
+ *
+ * @param {Number} n
+ *
+ * @returns {Array.<Number>}
+ */
+function generateArrayOfRandomNumbers(n) {
+  var array = [];
+
+  for (var i = 0; i < n; i++) {
+    array.push(getRandomNumber());
+  }
+
+  return array;
+}
+
+/**
+ * Generates n global suites (Mocha's describe)
+ *
+ * @param {Number} n
+ *
+ * @returns {Array.<Suite>}
+ */
+function generateNGlobalSuites(n) {
+  var suites = [rootSuite];
+
+  for (var i = 0; i < n; i++)  {
+    suites.push(objectAssign({}, globalSuite, {tests: []}));
+  }
+
+  return suites;
+}
+
+/**
+ * Generates n neste suites (Mocha's describe)
+ *
+ * @param {Number} n
+ *
+ * @returns {Array.<Suite>}
+ */
+function generateNNestedSuites(n) {
```

Comment:
This is actually constructing an expected result, right? These functions need better names...